tests/foreman/api/test_repository.py

```diff
@@ -380,10 +380,12 @@ def test_positive_create_checksum(self):
         :CaseImportance: Critical
         """
         for checksum_type in 'sha1', 'sha256':
-            with self.subTest(checksum_type):
-                repo = entities.Repository(
-                    product=self.product, checksum_type=checksum_type).create()
-                self.assertEqual(checksum_type, repo.checksum_type)
+            for downloadpolicy in 'background', 'immediate':
+                with self.subTest(checksum_type):
+                    repo = entities.Repository(
+                        product=self.product, checksum_type=checksum_type,
+                        download_policy=downloadpolicy).create()
+                    self.assertEqual(checksum_type, repo.checksum_type)
 
```

Comment:
adding an assertion for download policy as well make sense? what do you think?

Reply:
I think not requires , because if something fails around download policy the repocreate would also fail too.
Another side we are also having tests around only download policy:

https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/api/test_repository.py#L187

Reply:
that test case is specifically talking about checksum, so it is invalid to mix different functionalities

Reply:
basically, why did you think in that direction in the first place?

Reply:
@oshtaier earlier the test was creating the repo with default download_policy (on_demand) but not covering the other type of download_policies i.e. background/immediate hence added the same to cover  more on it.
and we added the nagative test for on_demand separately to cover on that type of download policy . 

Do you think here should be two test one per download policy background/immediate?

Reply:
nope, we need two test cases for different functionalities: one for checksum and one for download policies, so one test case should be left as it was before and check only checksum and one need check all types of download policies